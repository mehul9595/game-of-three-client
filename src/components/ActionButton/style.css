.action-button-container {
    display: flex;
    justify-content: space-around;
    width: 100%;
    margin: 20px 0;
}

.action-button-container .ant-btn.ant-btn-primary{
    width: 50px;
    height: 50px;
}