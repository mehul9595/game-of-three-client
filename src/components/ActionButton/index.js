import React from "react";
import { Button } from "antd";
import "antd/dist/antd.css";
import "./style.css";

const ActionButton = (props) => {
  return (
    <>
      <div className="action-button-container">
        <Button type="primary" shape="circle" size="large" value={-1} onClick={()=> props.actionHandler(-1)}>
          -1
        </Button>
        <Button type="primary" shape="circle" size="large" value={0} onClick={()=> props.actionHandler(0)}>
          0
        </Button>
        <Button type="primary" shape="circle" size="large" value={+1} onClick={()=> props.actionHandler(+1)}>
          +1
        </Button>
      </div>
    </>
  );
};

export default ActionButton;
