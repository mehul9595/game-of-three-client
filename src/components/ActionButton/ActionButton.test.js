import React from "react";
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';

// import "@testing-library/jest-dom";
import ActionButton from "./index";

describe("ActionButton Component", () => {
  test("should contain 3 action buttons", () => {
    const { getAllByRole } = render(<ActionButton />);

    expect(getAllByRole("button")).toBeTruthy();
    expect(getAllByRole("button").length).toBe(3);
  });

  test("should contain +1 action", () => {
    const { getByText } = render(<ActionButton />);
    
    expect(getByText("+1")).toHaveTextContent("+1");
  });

  test("should match the snapshot sucessfully", () => {
    const tree = renderer.create(<ActionButton />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
