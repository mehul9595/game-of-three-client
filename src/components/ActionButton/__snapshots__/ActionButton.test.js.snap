// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton Component should match the snapshot sucessfully 1`] = `
<div
  className="action-button-container"
>
  <button
    className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
    onClick={[Function]}
    type="button"
    value={-1}
  >
    <span>
      -1
    </span>
  </button>
  <button
    className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
    onClick={[Function]}
    type="button"
    value={0}
  >
    <span>
      0
    </span>
  </button>
  <button
    className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
    onClick={[Function]}
    type="button"
    value={1}
  >
    <span>
      +1
    </span>
  </button>
</div>
`;
